Server-side metrics docs (#1411)
grafana/tempo: docs/tempo/website/server_side_metrics/_index.md

---
title: Server-side metrics
weight: 400
---

# Server-side metrics

> Server-side metrics is a feature that allows to derive metrics from ingested traces.

Reviewer: ```suggestion
Server-side metrics is a feature that allows to derive metrics from ingested traces.
```

To generate metrics, it uses an additional component: the **metrics-generator**.
If present, the distributor will write received spans to both the ingester and the metrics-generator.
The metrics-generator processes spans and writes metrics to a Prometheus datasource using the Prometheus remote write protocol.

## Overview

Server-side metrics leverages the data available in Tempo's ingest path to provide additional value by generating metrics from traces.

> The metrics-generator internally runs **processors**.

Reviewer: ```suggestion
The metrics-generator internally runs a set of **processors**.
```

> Processors ingest span batches and keep track of metrics.

Reviewer: ```suggestion
Each processor ingests spans and produces metrics.
```

> Every processor derives different metrics, each providing a functionality.

Reviewer: ```suggestion
Every processor derives different metrics. Currently the following processors are available:
- Service graphs
- Span metrics
```

### Service graphs

> Service graphs are the representations of the relationships between services within a distributed system.

Reviewer: See suggestions above.
